main/main_timer_sync.cpp

```diff
@@ -230,40 +230,50 @@ bool MainTimerSync::DeltaSmoother::fps_allows_smoothing(int p_delta) {
 	return _measurement_allows_smoothing;
 }
 
-int MainTimerSync::DeltaSmoother::smooth_delta(int p_delta) {
+uint64_t MainTimerSync::DeltaSmoother::smooth_delta(uint64_t p_delta64) {
+	// Very important, ignore long deltas and pass them back unmodified.
+	// This is to deal with resuming after suspend for long periods.
+	// Delta is measured in ns, and will overflow with 32 bit integer after approx 36 mins.
+	if (p_delta64 > 1000000) {
+		return p_delta64;
+	}
```

Comment:
In theory it could overflow and be unlucky enough to land in the `[0; 1000000[` range, no?

Reply:
Discussed on chat, should be fine. It's the 32-bit int which is at risk of overflowing, not `p_delta64`.